src/roswire/proxy/roslaunch/reader.py

```diff
@@ -306,9 +309,22 @@ def _load_node_tag(self,
         nested_tags = [t for t in tag if t.tag in allowed]
         ctx_child, cfg = self._load_tags(ctx_child, cfg, nested_tags)
 
+        # locate node executable and determine the type
+        executable_path = self.locate_node_binary(package, name)
+        try:
```

Comment:
It would be neat if we pulled out lines 314 to 321 into a separate private method. (I.e., one that takes a given executable path and returns an `ExecutableType`).